tests/test_opt_solvers.py

```diff
@@ -68,8 +68,8 @@ def test_ipopt(self):
         for x_bad in [np.inf, np.nan]:
             x[n/2] = x_bad
             bad_prob = opt.opt_solver.QuadProblem(H,g,A,b,l,u,x=x)
-            Ipopt.solve(bad_prob)
-            self.assertEqual(Ipopt.get_status(),'error')
+            self.assertRaises(opt.opt_solver.opt_solver_error.OptSolverError_Ipopt, Ipopt.solve, bad_prob)
```

Comment:
You see, ipopt throws an OptSolverError_Ipopt when given nan or inf but there is not informative error message. Let's enhance this error exception class to store ipopt failure messages for all errors. This can be done on a separate PR.

Reply:
Okay. Thanks for fixing that. Is it possible to capture ipopt's error message?

Reply:
It should be possible. I will merge this PR and create a separate issue for capturing ipopt error messages and saving them in OptSolverError_Ipopt exception